scripts/test.sh

```diff
@@ -44,16 +48,33 @@ stop_server() {
   fi
 }
 
+startedSimulator=false
+log_temp=
+test_temp_dir=
 cleanup() {
   # Kill started object server
-  stop_server
-
+  stop_server || true
+  
+  # Give the simulator a chance to go down gracefully
+  if $startedSimulator; then
+    osascript -e 'tell app "Simulator" to quit without saving' || true
+    sleep 0.25 # otherwise the pkill following will get it
```

Comment:
Shouldn't this instead run a `pgrep` loop as was previously used in `reset-simulators.sh` to wait until the process is truly gone? Otherwise, this 0.25s wait seems very fragile, especially when running on resource constrained VMs under pressure...